src/services/firebase.ts

```diff
@@ -335,7 +336,33 @@ export namespace ourFirebase {
   }
 
   // Dispatches setSignals.
-  // TODO: export function listenToSignals() {}
+  export function listenToSignals() {
+    checkFunctionIsCalledOnce('listenToSignals');
+    getRef(`/gamePortal/gamePortalUserId/privateButAddable/signals`).on(
+      'value',
+      snap => {
+        if (!snap) {
+          return;
+        }
+        const signalsFbr: fbr.Signals = snap.val();
+        if (!signalsFbr) {
+          return;
+        }
+        const signals: SignalEntry[] = [];
+        Object.keys(signalsFbr).forEach(entryId => {
+          const signalFbr: fbr.SignalEntry = signalsFbr[entryId];
+          const signal: SignalEntry = {
```

Comment:
fbr.SignalEntry and SignalEntry  are identical, so you can just do:
const signal: SignalEntry = signalFbr;